controllers/velero.go

```diff
@@ -720,7 +720,7 @@ func (r DPAReconciler) noDefaultCredentials(dpa oadpv1alpha1.DataProtectionAppli
 		// go through cloudprovider plugins and mark providerNeedsDefaultCreds to false
 		for _, provider := range dpa.Spec.Configuration.Velero.DefaultPlugins {
 			if psf, ok := credentials.PluginSpecificFields[provider]; ok && psf.IsCloudProvider {
-				providerNeedsDefaultCreds[psf.PluginName] = needDefaultCred
+				providerNeedsDefaultCreds[string(provider)] = needDefaultCred
```

Comment:
This fixes the provider name used from `velero-plugin-for-gcp` to `gcp` which is what functions using value returned by this function expected.